### ISSUE TITLE
State of Support for Synology Platform

### ISSUE THREAD

**Issue by NorthernMan54** [oznu/homebridge-syno-spk#143]
Everyone,

If you have been following along recently, we have been challenged to support the platform and frequently produce releases that break the platform for everyone.  So we are asking the community for help, and are looking for someone to take over as SME for the platform, to manage releases and issues for the package.   I have been doing this for the last few months, and struggle as I do not have the technology or access to it.

From a time investment, is is likely a few hours a month, to publish updates when the core packages are refreshed, and to drive the resolution of issues with the community.  Basically it would involve troubleshooting issues, and working the homebridge collaborators when upstream issues occur.

If your interested, please let us know.

Thank you.

**Comment by paul-eeoc** [comment]
@NorthernMan54 
What skills are required?  I have a Synology server.  I could be a guinea pig and maintain this repository, if I have the skills required

**Comment by NorthernMan54** [comment]
@paul-eeoc Mostly it would be just testing and participating when a update is made to the core packages to ensure nothing is broken.

**Comment by BrentWilliams76** [comment]
I've been using this Synology app for a while now and find it invaluable.  I am not a developer, per se, but I don't mind contributing my time where I can.  if I had seen this post earlier, I would have offered sooner; but it was the upgrade 4.0.0 that caused me to look at "Issues" and see this for the first time.

**Comment by dyung** [comment]
I also have been using this for a while and would be willing to help out if I can. I was also unaware of the need for help, so let me know how I can help out!

**Comment by NorthernMan54** [comment]
What we are looking for is someone with access to a synology device, and able to do basic testing and troubleshooting.  Familiarity with GitHub would be an asset so that the text content could be updated as needed,

**Comment by NorthernMan54** [comment]
Goal would be for the next release, have it quickly validated, and rolled back if needed.

**Comment by BrentWilliams76** [comment]
Count me in.  Happy to help.

**Comment by dyung** [comment]
Also happy to help. I've got a DS1019+.

**Comment by NorthernMan54** [comment]
Tks everyone for the offer to help

The current release process just bundles the latest homebridge and homebridge-UI into a release archive and publishes it to the world.  I think we need to create an intermediate manual validation phase, in the release process so that a release can be checked manually before publishing.

I’m not a synology SME so bear with me

Question - what kind of artifact can we generate that could be easily tested and validated?  If you look at the GitHub releases - https://github.com/homebridge/homebridge-syno-spk/releases

Is it feasible to take the appropriate release asset and install it on your device?

If that it feasible, we could split the release process into two steps - one to generate the release assets and another to publish post validation.

**Comment by BrentWilliams76** [comment]
Yes, I think the .spk file is all we need to be able to test.  Naturally, we can only test the relevant release file for our system architecture (Arm, x86/braswell, etc.) but at least in broad terms of "does the release appear to run and be accessible on that architecture?" I think we should be able to answer for you.

**Comment by dyung** [comment]
I can confirm that I was able to install the homebridge-braswell-4.0.1.spk package on my DS1019+ and from a quick check things seem to be working.

**Comment by NorthernMan54** [comment]
Tks will wait for a couple more confirmations, then will publish.

**Comment by mitchoklemen** [comment]
I can also confirm that I was able to install the homebridge-braswell-4.0.1.spk package on my DS916+.
Node is still 20.13 so some plugins still dont work.

**Comment by BrentWilliams76** [comment]
Also was able to install 4.0.1.  No errors related to the core Homebridge instance or UI.  Seeing errors with one of my plug-ins (homebridge-zp for SONOS speakers) but they appear to be minor and most probably related to the plug-in itself.  Note that my Node.js is 20.9 so that is likely part of the issue.  I recommend to publish when ready.

**Comment by NorthernMan54** [comment]
Tks for all the confirmations, now to find a window to do the release when I’m available to do a rollback, if needed.

**Comment by NorthernMan54** [comment]
And just published…..fingers crossed

**Comment by NorthernMan54** [comment]
Question - would is make sense to change to a docker based approach going forward, so that the NodeJS version issue is removed ?

**Comment by dyung** [comment]
> Question - would is make sense to change to a docker based approach going forward, so that the NodeJS version issue is removed ?

How would this look in practice? I run a Homebridge instance in Docker on an older Synology and updating always seems a little tricky to me (but it may just be due to my lack of experience doing updates for it). I do like the current method where I can just use the Synology package manager UI to download and update the installed package. But if that is the direction most people want to go, I wouldn't be opposed to it as long as updates are relatively easy.

**Comment by dyung** [comment]
> Tks for all the confirmations, now to find a window to do the release when I’m available to do a rollback, if needed.

Do you have a plan for how future releases will work? Asking people to test and provide feedback in this issue might get a bit cumbersome. Might I suggest for the next release, perhaps open an issue for testing it, assign it to the testers, and then we can keep the discussion and any problems localized to that issue. This was just a quick thought, but I'm open to other ways of managing this if you know any better ones. (I would just like notifications without having to scan through a long thread to find out if there is anything I need to do!)

**Comment by moleribal** [comment]
I got the 4.0.1 update via DSM package manager and it works on DS1621+ (AMD). Thank you!

**Comment by mitchoklemen** [comment]
Can NODEjs 20.18 be compiled on synology SDK? If someone had done it already or can do it, can be put online like 3rdparty repo. Or I'm I wrong? Is this hard to do?

**Comment by 3Fingers** [comment]
4.01 installed and appears to have fixed issue encountered with 4.00. DS 1522+ w/AMD Ryzen R1600

**Comment by mitchoklemen** [comment]
I used mreidtt's compiled ver. for x64,
https://github.com/SynoCommunity/spksrc/actions/runs/13424166942/artifacts/2620615929

... installed it and no problems until, but ...

Homebridge doesn't recognize this ver. 22.14.0, strange, dont know why. Did restart DS and Homebridge.
Can this be solved?

...
The plugin "homebridge-..." requires a Node.js version of ^18.13.0 || ^20.16.0 || ^22.5.1 which does not satisfy the current Node.js version of v20.13.0. You may need to upgrade your installation of Node.js - see https://homebridge.io/w/JTKEF

He did packages for other also:
https://github.com/SynoCommunity/spksrc/actions/runs/13424166942

**Comment by mitchoklemen** [comment]
NorthernMan54  - Can you change the future NodeJS 22.14 acceptable version in homebridge?
Can this be done?
Thank you

**Comment by mitchoklemen** [comment]
https://archive.synology.com/download/Package/Node.js_v22/22.13.1-1004

Official Nodejs ver 22.13.1 is out

Can we change the script?

by:
https://github.com/hgy59
mandatory change:

you have to modify the source.sh script.
replace lines 7 to 15 with:
NODE_BIN_PATH="/var/packages/nodejs_v22/target/usr/local/bin:$NODE_BIN_PATH"
export HB_SERVICE_NODE_EXEC_PATH="${NODE_BIN_PATH}/node"

**Comment by arcusic** [comment]
I'd be happy to help out where I can as well.  Have a Synology RS822+.

**Comment by NorthernMan54** [comment]
FYI - I have documented the current release process

https://github.com/homebridge/documentation/blob/latest/Repositories.md#homebridge-syno-spk

1. Github Action - `Stage 1 - Build and Create Pre Release` Manually start the workflow and enter a new release tag ie 1.0.46
Note: Version number must not contain a v, ie 1.1.0 and not v1.1.0
2. Request the `Homebridge Synology Supporters` to validate the Pre Release Package. To do this create an Issue in the `homebridge/homebridge-syno-spk` repository and request the members of the github team `homebridge/homebridge-synology-supporters` to validate the release.
3. After a successfuly validation, publish the release with the github Action `Stage 2 - Publish Release` and enter the tag of release created in step 1.

To rollback/unpublish a release, use the GitHub action `Stage 2 - Publish Release` and enter the tag of a previous release.
